Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -177,46 +177,45 @@ function Publish-UpgradedServiceFabricApplication
         return
     }
 
-    if ($Action.Equals('RegisterAndUpgrade') -or $Action.Equals('Register'))
```

Comment:
Didn't removed. 
Moved this condition to downside.

Reply:
Why even move it?

Reply:
We are moving it downside so that we can decide weather application is already registered or not. If it is registered then there is no need to Unregister (followed by RegisterAndUpgrade), instead we can directly go ahead with Upgrade only. 
You can see in Line 215 of updated code that we are updating the status to Upgrade in case if application is already registered. Post this there will be If Condition (which I moved downside) which will work according to status change.

Reply:
To support the Upgrade $Action (as per my comment below) you can also just leave the existing if and move the body of your new if into an else for if ($reg)